syntax = "proto3";
option cc_enable_arenas = true;

package NFq.NConfig;
option java_package = "ru.yandex.kikimr.proto";

import "ydb/core/fq/libs/config/protos/audit.proto";
import "ydb/core/fq/libs/config/protos/checkpoint_coordinator.proto";
import "ydb/core/fq/libs/config/protos/common.proto";
import "ydb/core/fq/libs/config/protos/compute.proto";
import "ydb/core/fq/libs/config/protos/control_plane_proxy.proto";
import "ydb/core/fq/libs/config/protos/control_plane_storage.proto";
import "ydb/core/fq/libs/config/protos/db_pool.proto";
import "ydb/core/fq/libs/config/protos/gateways.proto";
import "ydb/core/fq/libs/config/protos/health_config.proto";
import "ydb/core/fq/libs/config/protos/nodes_manager.proto";
import "ydb/core/fq/libs/config/protos/pending_fetcher.proto";
import "ydb/core/fq/libs/config/protos/pinger.proto";
import "ydb/core/fq/libs/config/protos/private_api.proto";
import "ydb/core/fq/libs/config/protos/private_proxy.proto";
import "ydb/core/fq/libs/config/protos/quotas_manager.proto";
import "ydb/core/fq/libs/config/protos/rate_limiter.proto";
import "ydb/core/fq/libs/config/protos/read_actors_factory.proto";
import "ydb/core/fq/libs/config/protos/resource_manager.proto";
import "ydb/core/fq/libs/config/protos/test_connection.proto";
import "ydb/core/fq/libs/config/protos/token_accessor.proto";
import "ydb/library/folder_service/proto/config.proto";

////////////////////////////////////////////////////////////

message TConfig {
    bool Enabled = 1;
    TCommonConfig Common = 2;
    TControlPlaneStorageConfig ControlPlaneStorage = 3;
    TControlPlaneProxyConfig ControlPlaneProxy = 4;
    NKikimrProto.NFolderService.TFolderServiceConfig FolderService = 5;
    TPrivateApiConfig PrivateApi = 6;
    TTokenAccessorConfig TokenAccessor = 7;
    TDbPoolConfig DbPool = 8;
    TGatewaysConfig Gateways = 9;
    TCheckpointCoordinatorConfig CheckpointCoordinator = 10;
    TAuditConfig Audit = 11;
    TResourceManagerConfig ResourceManager = 12;
    TPrivateProxyConfig PrivateProxy = 13;
    TNodesManagerConfig NodesManager = 14;
    TPendingFetcherConfig PendingFetcher = 15;
    TPingerConfig Pinger = 16;
    bool EnableDynamicNameservice = 17;
    TTestConnectionConfig TestConnection = 18;
    TReadActorsFactoryConfig ReadActorsFactoryConfig = 19;
    THealthConfig Health = 20;
    TQuotasManagerConfig QuotasManager = 21;
    TRateLimiterConfig RateLimiter = 22;
    bool EnableTaskCounters = 23;
    TComputeConfig Compute = 24;
}
